src/Storages/Distributed/DirectoryMonitor.cpp

```diff
@@ -420,6 +428,18 @@ void StorageDistributedDirectoryMonitor::run()
         {
             try
             {
+                if (!pool)
+                {
+                    pool = StorageDistributedDirectoryMonitor::createPool(name, storage);
+                }
+                else
+                {
+                    if (need_update_pool)
+                    {
+                        need_update_pool = false;
+                        pool = StorageDistributedDirectoryMonitor::createPool(name, storage);
+                    }
+                }
```

Comment:
This can be written in a single branch, i.e.

```suggestion
                if (!pool || need_update_pool)
                {
                    pool = StorageDistributedDirectoryMonitor::createPool(name, storage);
                    need_update_pool = false;
                }
```

Or if it was splitted, worth adding some log.